WordPress/gutenberg
# DataViews: update filters in view config

Part of https://github.com/WordPress/gutenberg/issues/55083

## What?

This PR updates how filters are stored in the view config.

Before:

```js
{
  filters: {
    author: 1,
    author_exclude: 2,
    status: [ 'publish', 'draft' ],
    status_exclude: [ 'scheduled' ] 
  }
}
```

After:

```js
{
  filters: [
    { field: 'author', operator: 'in', value: 1 },
    { field: 'author', operator: 'notIn', value: 2 },
    { field: 'status', operator: 'in', value: 'publish,draft' },
    { field: 'status', operator: 'notIn', value: 'scheduled' },
  ]
}
```

## Why?

This provides a general syntax that consumers can use to query their datasets.

## Testing Instructions

- Enable the "wp admin" experiment in "Gutenberg > Experiments".
- Go to "Appareance > Editor > Pages > Manage all pages".
- Verify the filters work as expected.

You may need to clear any previously stored dataview entity/CPT. One way to do it is to set `show_ui` to `true` in `dataviews.php` so they are listed in wp-admin. Then, delete it.
